### ISSUE TITLE
when ZMQ_STREAM  recving data from a non-zmq node, what's the correct model it will follow ?

### ISSUE THREAD

**Issue by ppLorins** [zeromq/libzmq#2172]
Im using zmq_stream socket(as the client) for communicating with a non-zmq peer(as the server). The communicating procedure is exactly an RPC :

1. I send a payload to the server 
2. Server give me the response.

But sometimes the response size is very large , that I have to call zmq_msg_recv() for more than one time.
According to the zmq official document

```
When receiving TCP data, a ZMQ_STREAM socket shall prepend a message part containing the identity of the originating peer to the message before passing it to the application. Messages received are fair-queued from among all connected peers.
```

I should invoke  zmq_msg_recv() for receiving an identity frame every time before I invoke zmq_msg_recv() for the real payload frame.Thus , the functions calls I made will look like this(assume the server's response is large enough):

zmq_msg_recv(for id frame) --> zmq_msg_recv(for id real payload) --> zmq_msg_recv(for id frame) -->
zmq_msg_recv(for id real payload).....   [loop like this ]

Under my test(server give a 2000 bytes long response), every id frame has a 'msg more' flag in it , but the payload is not .

My question:  will it be possible that the frame which contains the real payload will  also has a 'more flag' in it ?   

I suppose it won't.  Cause zmq know nothing about the boundary of the stream data when using zmq_stream talking with a non-zmq node.  Thus it should give the payload everytime it can read from tcp buff to the application layer ,as a zmq_msg_t , and prepending it with an id frame. 

So , is my guess right ?

**Comment by bluca** [comment]
The more flag is an internal concept in ZMQ, related to multi-part messages. It is not and cannot be used to signal that there is more data on a socket, you can use zmq_poll for that.

I recommend reading the guide if you are more interested in such details: http://zguide.zeromq.org/page:all

**Comment by ppLorins** [comment]
@bluca     Yes I know that ,and I've dived into it quite a lot before. What my concern currently is  , what's the right action of zmq_stream socket for recving some big enough data,it has a big influence about my application's recv behaviour .
